### DISCUSSION TITLE
Elementor Entrance Animation | wait for the widget/container to be completely in the viewport to trigger the animation

### DISCUSSION THREAD

**Discussion by Bharat610** [elementor/elementor#23834] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

I am trying to do custom animations, where I am just using the elementor animation classes from the DOM to manipulate the animations, that way I don't have to use JS to detect when the widget/container is in the viewport. 

The only problem that I am facing is that the animation triggers as soon the widget/container touches the viewport it doesn't wait for them to be completely inside the viewport to trigger.

### Describe the solution you'd like

It would be nice if we can add a feature where we have the option to choose the triggering point for the animation, that is;

1) trigger the animation as soon as it touches the visible viewport (default functionality)
2) wait for the widget/container to be completely inside the visible viewport to trigger the animation (feature request)
3) custom distance from the viewport to trigger the animation (feature request)

### Describe alternatives you've considered

For now I am just adding 'animation delay' to delay the trigger but this isn't a viable option. Thanks!!

**Comment by datyorubadev** [comment]
Dear Elementor Team, we will be glad if this is implemented in the future updates, by adding interaction timeline as a feature of Elementor. So, we will be control and create our interaction just like Webflow.
